src/main/java/org/dependencytrack/resources/v1/WorkflowResource.java

```diff
@@ -0,0 +1,54 @@
+package org.dependencytrack.resources.v1;
+
+import alpine.common.logging.Logger;
+import alpine.server.auth.PermissionRequired;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import io.swagger.annotations.Authorization;
+import org.dependencytrack.auth.Permissions;
+import org.dependencytrack.model.WorkflowState;
+import org.dependencytrack.persistence.QueryManager;
+import org.dependencytrack.resources.v1.vo.IsTokenBeingProcessedResponse;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.List;
+import java.util.UUID;
+
+@Path("/v1/workflow")
+@Api(value = "workflow", authorizations = @Authorization(value = "X-Api-Key"))
+public class WorkflowResource {
+
+    private static final Logger LOGGER = Logger.getLogger(WorkflowResource.class);
+
+    @GET
+    @Path("/token/{uuid}/status")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Retrieves workflow states associated with the token received from bom upload .", response = IsTokenBeingProcessedResponse.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 401, message = "Unauthorized")
+    })
+    @PermissionRequired(Permissions.Constants.BOM_UPLOAD)
+    public Response isTokenBeingProcessed(
```

Comment:
Should be renamed to describe more accurately what it's doing.

```suggestion
    public Response getWorkflowStates(
```